### DISCUSSION TITLE
[Bug]: I can't properly filter data when selecting a specific date or time period

### DISCUSSION THREAD

**Discussion by anagonzaga** [Power-Components/livewire-powergrid#1055] [Q&A]
### Have you searched through other issues to see if your problem is already reported or has been fixed?

Yes, I did not find it.

### Did you read the documentation?

Yes, I did not find it.

### Have you tried to publish the views?

Yes - I didn't work.

### Is there an error in the console?

No

### PHP Version

8

### PowerGrid

3.7

### Laravel

8

### Livewire

2.10

### Alpine JS

3.0

### Theme

Tailwind 3.x with tailwind/forms

### Describe the bug.

I'm facing a problem with filtering the data in a specific column in my system. In this column, I used "makeInputDatePicker()" to add a date selection filter. However, when I select a single date or a period, the information is not being filtered correctly.

### To Reproduce...

First click on "FOO" then....

### Extra information

```markdown
<?php
    
    public function datasource(): \Illuminate\Support\Collection
    {
        return $this->vagas();
    }

    public function vagas() 
    {

        return [
            {
              "id": 1,
              "uuid": "d4e5f6g7",
              "data": "2023-07-10",
              "descricao_turno": "manhã"
            },
            {
              "id": 2,
              "uuid": "h8i9j0k1",
              "data": "2023-07-25",
              "descricao_turno": "tarde"
            },
            {
              "id": 3,
              "uuid": "l2m3n4o5",
              "data": "2023-07-01",
              "descricao_turno": "manhã"
            },
            {
              "id": 4,
              "uuid": "p6q7r8s9",
              "data": "2023-06-11",
              "descricao_turno": "tarde"
            },
            {
              "id": 5,
              "uuid": "t0u1v2w3",
              "data": "2023-06-12",
              "descricao_turno": "manhã"
            }
        ];

    }

    public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn('id')
            ->addColumn('uuid')

            ->addColumn('data_vaga_formatted', function($model){
                return Carbon::parse($model->data_vaga)->format('d/m/Y');
            
            })
            >addColumn('descricao_turno')      
    }


    public function columns(): array
    {

        return [
            Column::add()
                ->title('Data Vaga')
                ->field('data_vaga_formatted', 'data_vaga')
                ->searchable()
                ->sortable()
                ->makeInputDatePicker(),


            Column::add()
                ->title('Turno')
                ->field('descricao_turno', 'descricao_turno')
                ->searchable()
                ->sortable(),
        ]
    }
```

**Comment by luanfreitasdev** [comment]
Hello @anagonzaga , You must also put the date or date_vaga column in addColumns :  >addColumn('data') 

But data_formatted is wrong. Carbon should read from $model->data and not data_vaga

**Comment by anagonzaga** [comment]
Hello @luanfreitasdev, 


I'm sorry, I forgot to change the name date to data_vaga in the vacancy method to share the code, but I did that too, I changed the name and even then the datepicker doesn't filter..

**Comment by luanfreitasdev** [comment]
Any chance to use the latest version?. Perhaps there is a bug in the date filters in the previous version
